api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -238,3 +246,56 @@ func (r *OpenStackDataPlaneNodeSetSpec) duplicateNodeCheck(nodeSetList *OpenStac
 
 	return
 }
+
+// Compare TLS settings of control plane and data plane
+// if control plane name is specified attempt to retrieve it
+// otherwise get any control plane in the namespace
+func (r *OpenStackDataPlaneNodeSetSpec) ValidateTLS(namespace string) error {
+	var err error
+
+	controlPlane := openstackv1.OpenStackControlPlane{}
+	if len(r.ControlPlane) > 0 {
+		controlPlaneName := types.NamespacedName{
+			Name:      r.ControlPlane,
+			Namespace: namespace,
+		}
+		err = webhookClient.Get(context.TODO(), controlPlaneName, &controlPlane)
+		if err == nil {
+			err = r.TLSMatch(controlPlane)
+		}
+
```

Comment:
In the case where the control plane is not found, we should requeue and wait, right?

Reply:
ah good point.  I forgot about the no control plane case.  @slagle  - this is a legit case, right?